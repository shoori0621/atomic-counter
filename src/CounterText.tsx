import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

interface Props {
  count: number;
}

export default function CounterText(props: Props) {
  const {count} = props;
  return (
    <View style={styles.textContainer}>
      {count >= 10 && <Text>you're great!</Text>}
      <Text style={styles.text}>{count}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  textContainer: {
    width: 100,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: 20,
  },
});
