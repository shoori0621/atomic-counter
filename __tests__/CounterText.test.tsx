import React from 'react';
import renderer from 'react-test-renderer';
import CounterText from '../src/CounterText';

// Note: import explicitly to use the types shiped with jest.
import {it, describe, expect} from '@jest/globals';

describe('CounterText', () => {
  it('通常のcounterTextを描画できる', () => {
    const component = renderer.create(<CounterText count={0} />);
    expect(component).toMatchSnapshot();
  });
  it('カウントが10以上のときのテキストが表示されているCounterTextを描画できる', () => {
    const component = renderer.create(<CounterText count={10} />);
    expect(component).toMatchSnapshot();
  });
});
