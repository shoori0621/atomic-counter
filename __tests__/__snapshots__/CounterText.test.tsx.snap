// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterText カウントが10以上のときのテキストが表示されているCounterTextを描画できる 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "width": 100,
    }
  }
>
  <Text>
    you're great!
  </Text>
  <Text
    style={
      {
        "fontSize": 20,
      }
    }
  >
    10
  </Text>
</View>
`;

exports[`CounterText 通常のcounterTextを描画できる 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "width": 100,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 20,
      }
    }
  >
    0
  </Text>
</View>
`;
